### ISSUE TITLE
try/catch options are used too often when parameter checks would be better

### ISSUE THREAD

**Issue by garywade** [tomaz/appledoc#594]
One example where try/catch is used when a parameter check would be better is in GBCommentsProcessor.m in the method processAbstractBlockInString:lines:blockRange:shortRange: which becomes very noisy when doing debugging with Objective-C try/catch exceptions turned on.

This original code:

	NSRange range;
	@try {
		description = [description substringToIndex:index.location];
	}
	@catch (NSException *exception) {
		
	}
	@finally {
		range = [string rangeOfString:description];
	}
	NSString *prefix = nil;

Could become this code:

	NSRange range;
	if (index.location != NSNotFound)
	{
		description = [description substringToIndex:index.location];
	}
	range = [string rangeOfString:description];
	NSString *prefix = nil;

**Comment by tomaz** [comment]
Yeah, it's quite an old codebase, there are many code cleanup opportunities... If you feel for it, feel free to create pull request, just make sure the functionality remanes intact - add unit tests where missing...

Also for above example, it might be better `if (index.location < description.length)` to guard against index out of bounds (didn't check surrounding code though, so it may not be needed, just wanted to point out possible issue with that change which was covered before)
